SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -146,225 +146,284 @@ void ComponentReader::ApplyComponentUpdate(const Worker_ComponentId ComponentId,
-						const bool bHasReferences = NewUnresolvedRefs.Num() > 0 || NewMappedRefs.Num() > 0;
-
-						if (ReferencesChanged(RootObjectReferencesMap, SwappedCmd.Offset, bHasReferences, NewMappedRefs, NewUnresolvedRefs))
+						else
 						{
-							if (bHasReferences)
-							{
-								RootObjectReferencesMap.Add(
-									SwappedCmd.Offset,
-									FObjectReferences(ValueData, CountBits, MoveTemp(NewMappedRefs), MoveTemp(NewUnresolvedRefs),
-													  ShadowOffset, Cmd.ParentIndex, ArrayProperty, /* bFastArrayProp */ true));
-							}
-							else
-							{
-								RootObjectReferencesMap.Remove(SwappedCmd.Offset);
-							}
-							bOutReferencesChanged = true;
+							RootObjectReferencesMap.Remove(SwappedCmd.Offset);
 						}
+						ApplySchemaObjectData.bOutReferencesChanged = true;
 					}
-					else
-					{
-						ApplyArray(ComponentObject, FieldId, RootObjectReferencesMap, ArrayProperty, Data, SwappedCmd.Offset, ShadowOffset,
-								   Cmd.ParentIndex, bOutReferencesChanged);
-					}
 				}
 				else
 				{
-					ApplyProperty(ComponentObject, FieldId, RootObjectReferencesMap, 0, Cmd.Property, Data, SwappedCmd.Offset, ShadowOffset,
-								  Cmd.ParentIndex, bOutReferencesChanged);
+					ApplyArray(ApplySchemaObjectData.ComponentObject, FieldId, RootObjectReferencesMap, ArrayProperty, Data,
+							   SwappedCmd.Offset, ShadowOffset, Cmd.ParentIndex, ApplySchemaObjectData.bOutReferencesChanged);
+				}
+			}
+			else
+			{
+				// The ReplicatedMovement struct is a special case because it contains bRepPhysics, we read just that bit.
+				if (bIsSpecialCaseProperty && (int32)AActor::ENetFields_Private::ReplicatedMovement == Cmd.ParentIndex)
+				{
+					TArray<uint8> ValueData = IndexBytesFromSchema(ApplySchemaObjectData.ComponentObject, FieldId, 0);
+					Channel.Actor->GetReplicatedMovement_Mutable().bRepPhysics = ValueData[0] & (1 << 1) ? true : false;
+				}
+				else
+				{
+					ApplyProperty(ApplySchemaObjectData.ComponentObject, FieldId, RootObjectReferencesMap, 0, Cmd.Property, Data,
+								  SwappedCmd.Offset, ShadowOffset, Cmd.ParentIndex, ApplySchemaObjectData.bOutReferencesChanged);
 				}
+			}
 
-				if (Cmd.Property->GetFName() == NAME_RemoteRole)
+			if (Cmd.Property->GetFName() == NAME_RemoteRole)
+			{
+				// Downgrade role from AutonomousProxy to SimulatedProxy if we aren't authoritative over
+				// the client RPCs component.
+				GDK_PROPERTY(ByteProperty)* ByteProperty = GDK_CASTFIELD<GDK_PROPERTY(ByteProperty)>(Cmd.Property);
+				if (ByteProperty->GetPropertyValue(Data) == ROLE_AutonomousProxy)
 				{
-					// Downgrade role from AutonomousProxy to SimulatedProxy if we aren't authoritative over
-					// the client RPCs component.
-					GDK_PROPERTY(ByteProperty)* ByteProperty = GDK_CASTFIELD<GDK_PROPERTY(ByteProperty)>(Cmd.Property);
-					if (!bIsAuthServer && !bAutonomousProxy && ByteProperty->GetPropertyValue(Data) == ROLE_AutonomousProxy)
+					Channel.SetAutonomousProxyOnAuthority(true);
```

Comment:
Potential edge case with this: if the server later decides to set autonomous proxy to false, and then change the ownership to a different client (that had the actor checked out before), this would still be true on that client so it would set its role to autonomous proxy.

I think doing this outside of the if instead would prevent that:
```
Channel.SetAutonomousProxyOnAuthority(ByteProperty->GetPropertyValue(Data) == ROLE_AutonomousProxy);
```

Reply:
Done!